In [0]:
import pandas as pd
import arrow as arrow
import statsmodels.api as sm

DATA_URL = 'http://128.138.93.164/aprd6342/data/TravelPonyFacebook.csv'
df = pd.read_csv(DATA_URL)

In [0]:
df.head()

Start Date  ... Cost per Page Engagement (USD)
0  2013-12-02  ...                       1.219512
1  2013-12-01  ...                       1.395349
2  2013-11-30  ...                       1.190476
3  2013-11-29  ...                       1.167315
4  2013-11-22  ...                       2.512857

[5 rows x 20 columns]

In [0]:
df['CPI'] = df['Amount Spent (USD)']/df['Impressions']
df.head()

Start Date    End Date  ... Cost per Page Engagement (USD)       CPI
0  2013-12-02  2013-12-02  ...                       1.219512  0.000466
1  2013-12-01  2013-12-01  ...                       1.395349  0.000652
2  2013-11-30  2013-11-30  ...                       1.190476  0.000448
3  2013-11-29  2013-11-29  ...                       1.167315  0.000452
4  2013-11-22  2013-11-22  ...                       2.512857  0.002464

[5 rows x 21 columns]

In [0]:
!pip install arrow

     |████████████████████████████████| 51kB 1.7MB/s 


In [0]:
df['Start Date'] = pd.to_datetime(df['Start Date'])
df['Day of Week'] = df['Start Date'].dt.day_name()

df.groupby('Day of Week',as_index=False).agg({'CPI': 'sum'})

Day of Week       CPI
0      Friday  2.056639
1      Monday  1.713976
2    Saturday  1.311720
3      Sunday  1.755447
4    Thursday  1.950631
5     Tuesday  1.440979
6   Wednesday  1.761665

In [0]:
print(df[['Amount Spent (USD)','Reach','Frequency','Unique Clicks','Page Likes']].corr())

                    Amount Spent (USD)     Reach  ...  Unique Clicks  Page Likes
Amount Spent (USD)            1.000000  0.703124  ...       0.882993    0.757612
Reach                         0.703124  1.000000  ...       0.722249    0.304388
Frequency                     0.130201  0.334101  ...       0.135103    0.000182
Unique Clicks                 0.882993  0.722249  ...       1.000000    0.584614
Page Likes                    0.757612  0.304388  ...       0.584614    1.000000

[5 rows x 5 columns]


In [0]:
model1 = sm.OLS(df['Unique Clicks'], df['Reach']+df['Frequency']).fit()
model1.summary()



<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:          Unique Clicks   R-squared (uncentered):                   0.089
Model:                            OLS   Adj. R-squared (uncentered):              0.089
Method:                 Least Squares   F-statistic:                              363.1
Date:                Thu, 14 Nov 2019   Prob (F-statistic):                    2.64e-77
Time:                        03:07:56   Log-Likelihood:                         -17307.
No. Observations:                3705   AIC:                                  3.462e+04
Df Residuals:                    3704   BIC:                                  3.462e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Frequency      7.9291      0.416     19.056      0.000       7.113       8.745
==============================================================================
Omnibus:                     5923.176   Durbin-Watson:                   0.694
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          3853539.229
Skew:                          10.278   Prob(JB):                         0.00
Kurtosis:                     159.651   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [0]:
model2 = sm.OLS(df['Unique Clicks'], df['Reach']).fit()
model2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:          Unique Clicks   R-squared (uncentered):                   0.539
Model:                            OLS   Adj. R-squared (uncentered):              0.539
Method:                 Least Squares   F-statistic:                              4334.
Date:                Thu, 14 Nov 2019   Prob (F-statistic):                        0.00
Time:                        03:10:03   Log-Likelihood:                         -16045.
No. Observations:                3705   AIC:                                  3.209e+04
Df Residuals:                    3704   BIC:                                  3.210e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Reach          0.0020   3.09e-05     65.832      0.000       0.002       0.002
==============================================================================
Omnibus:                     4800.065   Durbin-Watson:                   0.762
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          5192547.294
Skew:                           6.420   Prob(JB):                         0.00
Kurtosis:                     185.951   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [0]:
model3 = sm.OLS(df['Unique Clicks'], df['Frequency']).fit()
model3.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:          Unique Clicks   R-squared (uncentered):                   0.089
Model:                            OLS   Adj. R-squared (uncentered):              0.089
Method:                 Least Squares   F-statistic:                              363.1
Date:                Thu, 14 Nov 2019   Prob (F-statistic):                    2.64e-77
Time:                        03:10:05   Log-Likelihood:                         -17307.
No. Observations:                3705   AIC:                                  3.462e+04
Df Residuals:                    3704   BIC:                                  3.462e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Frequency      7.9291      0.416     19.056      0.000       7.113       8.745
==============================================================================
Omnibus:                     5923.176   Durbin-Watson:                   0.694
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          3853539.229
Skew:                          10.278   Prob(JB):                         0.00
Kurtosis:                     159.651   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [0]:
y = df['Unique Clicks']
x = df[['Reach','Frequency']]

model4 = sm.OLS(y, x).fit()
model4.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:          Unique Clicks   R-squared (uncentered):                   0.557
Model:                            OLS   Adj. R-squared (uncentered):              0.556
Method:                 Least Squares   F-statistic:                              2325.
Date:                Thu, 14 Nov 2019   Prob (F-statistic):                        0.00
Time:                        03:08:54   Log-Likelihood:                         -15973.
No. Observations:                3705   AIC:                                  3.195e+04
Df Residuals:                    3703   BIC:                                  3.196e+04
Df Model:                           2                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Reach          0.0019   3.12e-05     62.490      0.000       0.002       0.002
Frequency      3.6139      0.298     12.109      0.000       3.029       4.199
==============================================================================
Omnibus:                     5107.616   Durbin-Watson:                   0.803
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          5128392.747
Skew:                           7.331   Prob(JB):                         0.00
Kurtosis:                     184.674   Cond. No.                     9.84e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 9.84e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

1) The worst day of the week in terms of CPI is Friday at 2.056639. Saturday is the cheapest to generate impressions with 1.311720.


2) Unique Clicks has the strongest correlation with Amount Spent at .88. This means amount spent and unique clicks have a strong positive relationship.


3)The model indicates that reach more strongly predicts Unique Clicks it has a much larger T value and when run seperately the adjusted R squared is much higher. A Unique Click can only happen once so Frequency isnt as applicable.